### PR TITLE
fix(diagnostics): read spectrum data from the per-AP route

### PR DESCRIPTION
**Pull request by jpennin5**
## Description

**Problem:** `get_spectrum_scan` and `list_spectrum_interference` query a
site-wide `stat/spectrumscan` path that does not exist — every call 404s.

**Root cause + fix:** spectrum scans are per access point:
`stat/spectrum-scan/{ap_mac}`, verified live against Network 10.5.67 —
each AP answers with a `scans[]` list per radio carrying a
`spectrum_table` plus scan-state flags (a wrong MAC fails with
`api.err.UnknownDevice`, confirming the route parses MACs).

- `get_spectrum_scan` queries one AP (`ap_mac`) or enumerates the site's
  APs, returning the controller's per-AP shape as reported.
- `list_spectrum_interference` flattens the `spectrum_table` rows
  annotated with AP MAC and radio, passing row fields through verbatim
  rather than guessing at a schema.
- An AP that has never run an RF scan reports empty tables, which is a
  state, not an error.

**Live verification (Network 10.5.67):** all site APs answer with
per-radio scan state.

## Type of Change

- [x] Bug fix (non-breaking change that fixes an issue)

The old path never returned anything; the new optional `ap_mac` parameter
defaults to all APs.

## Testing

- Fixtures mirror the real per-AP response shape; tests cover single-AP
  and enumerate-all paths, radio flattening/annotation, and the
  never-scanned empty state.
- `pytest tests/unit/` passes; pre-commit hooks pass for all files this
  PR touches.

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation (CHANGELOG.md)
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published (none — stands alone)

## AI Assistance

This PR was created with assistance from Claude Code.

**Human Review Status:** ✅ Reviewed and approved by @jpennin5
**Testing:** Full unit suite passing; per-AP route verified against a
live Network 10.5.67 controller.
**Security Review:** Completed — all fixture MACs are from the RFC 7042
documentation range.

### CODE REVIEW

File: src/tools/diagnostics.py [hunk 1]
```diff
@@ -138,73 +133,111 @@ async def get_speed_test_history(site_id: str, settings: Settings) -> list[dict[
...
+        client: Authenticated API client
+        site_id: Site identifier
+        ap_mac: Explicit AP MAC, or None to enumerate the site's APs
+
+    Returns:
+        List of AP MAC addresses
+    """
+    if ap_mac:
+        return [ap_mac]
+    response = await client.get(f"/ea/sites/{site_id}/devices")
```
[the remark this answers is not in the document]
Comment (jpennin5) [reply src/tools/diagnostics.py:208]: Fixed in f3c6b3d — `ap_mac` now goes through `validate_mac_address` like the sibling tools before it reaches the request path.

File: src/tools/diagnostics.py [hunk 2]
```diff
@@ -138,73 +133,111 @@ async def get_speed_test_history(site_id: str, settings: Settings) -> list[dict[
...
+                continue
+            for row in radio_scan.get("spectrum_table", []) or []:
+                if isinstance(row, dict):
+                    entries.append(
+                        {
+                            "ap_mac": ap.get("mac"),
+                            "radio": radio_scan.get("radio"),
+                            "radio_name": radio_scan.get("name"),
+                            **row,
+                        }
```
[the remark this answers is not in the document]
Comment (jpennin5) [reply src/tools/diagnostics.py:300]: Fixed in f3c6b3d: the row merges first and the AP/radio annotations last, so a same-named controller key can never mask them. Regression test included.

**Comment by enuno** [comment]
### Summary
Request changes. The PR is not yet merge-ready because the current status checks are failing.

### Critical findings
- Merge state: \ / \
- Failing check(s): Mermaid Diagram Sync Assistant
- The changed tooling surface should be revalidated after the failing check is resolved.

### Suggestions
1. Resolve the failing status check(s) and rerun CI.
2. Reconfirm the changed tooling behavior against the intended controller/fixture shapes.
3. Ensure regression tests cover the original failure mode and edge cases introduced here.
4. Update docs if any request/response or error semantics changed.

### Verification checklist for the author
- [ ] Required checks are passing
- [ ] Regression tests cover the changed path
- [ ] Lint / format / type checks pass locally
- [ ] Documentation is updated where user-facing behavior changed
- [ ] No secrets, credentials, or controller-identifying data were introduced

**Review by enuno — CHANGES_REQUESTED** [review]
Merge blocked until the failing status checks are resolved and the tooling changes are revalidated.

**Comment by jpennin5** [comment]
The failing check is the `Mermaid Diagram Sync Assistant` GitHub App, not a workflow in this repository. Its check-run output on this PR reads: "Webhook Processing Failed — An unexpected error occurred while processing the webhook. Please try pushing a new commit or check your repository configuration." That processing happens on the app's side; pushing new commits to these PRs has re-triggered the check with the same result, so it is not reachable from a fork.

This is the loop documented in #112: the app has errored on every fork PR since late June, fork CI runs sit at `action_required` awaiting maintainer approval, and the daily automated review then reads the app error as a failing required check and requests changes.

Every check that can conclude on a fork PR is green here (both Socket Security checks). The remaining steps are repo-side — approving the CI runs and either repairing/removing the app integration or unrequiring its check. #117 merged with the identical check state, so the merge path exists. Happy to rebase or adjust anything code-side the moment CI can actually run.
